csi_camera.cpp

```diff
@@ -0,0 +1,438 @@
+            }
+                
+            if (attrs->count("height_px") == 1) {
+                std::shared_ptr<ProtoType> height_proto = attrs->at("height_px");
+                auto height_value = height_proto->proto_value();
+                if (height_value.has_number_value()) {
+                    int height_num = static_cast<int>(height_value.number_value());
+                    height_px = height_num;
+                }
+            }
+            if (!height_px) {
+                std::cout << "ERROR: height_px attribute not found" << std::endl;
+                std::cout << "Setting height_px to default value: " << DEFAULT_INPUT_HEIGHT << std::endl;
+                height_px = DEFAULT_INPUT_HEIGHT;
+            }
+            
+            if (attrs->count("frame_rate") == 1) {
+                std::shared_ptr<ProtoType> frame_proto = attrs->at("frame_rate");
+                auto frame_value = frame_proto->proto_value();
+                if (frame_value.has_number_value()) {
+                    int frame_num = static_cast<int>(frame_value.number_value());
+                    frame_rate = frame_num;
+                }
+            }
+            if (!frame_rate) {
+                std::cout << "ERROR: frame_rate attribute not found" << std::endl;
+                std::cout << "Setting frame_rate to default value: " << DEFAULT_INPUT_SENSOR<< std::endl;
+                frame_rate = DEFAULT_INPUT_FRAMERATE;
+            }
+            
+            if (attrs->count("video_path") == 1) {
+                std::shared_ptr<ProtoType> video_proto = attrs->at("video_path");
+                auto video_value = video_proto->proto_value();
+                if (video_value.has_string_value()) {
+                    std::string video_str = video_value.string_value();
+                    video_path = video_str;
+                }
+            }
+            if (video_path.empty() ) {
+                std::cout << "ERROR: video_path attribute not found" << std::endl;
+                std::cout << "Setting video_path to default value: " << DEFAULT_INPUT_SENSOR << std::endl;
+                video_path = DEFAULT_INPUT_SENSOR;
+            }
+
+            if (attrs->count("debug") == 1) {
+                std::shared_ptr<ProtoType> debug_proto = attrs->at("debug");
+                auto debug_value = debug_proto->proto_value();
+                if (debug_value.has_bool_value()) {
+                    bool debug_bool = static_cast<bool>(debug_value.bool_value());
+                    debug = debug_bool;
+                }
+            }
+        }
+
+        // OVERRIDE
+
+        void reconfigure(Dependencies deps, ResourceConfig cfg) override {
+            if (debug) {
+                std::cout << "Reconfiguring CSI Camera module" << std::endl;
+            }
+
+            // Update attributes
+            auto attrs = cfg.attributes();
+            validate_attrs(attrs);
+
+            // Stop gst pipeline
+            stop_pipeline();
```

Comment:
We'd only need to reconfigure if the resolution or frame format changes, no? Is that checked somewhere? I'm not 100% sure when this function will get called, but we wouldn't want to reconfigure this camera anytime _anything_ in the config changes.

Reply:
Good point. 

RDK will handle only sending `reconfigure` calls when the module or associated component params are changed.

If a change is made to the following params, GST changes are not required.
```
"name": "csicam_test",
"namespace": "rdk",
"type": "camera",
"debug": true,
"model": "viam:camera:csi"
```